tests/test_file_download.py

```diff
@@ -96,7 +102,17 @@ def test_revision_not_found(self):
             filename=CONFIG_NAME,
             revision=DUMMY_MODEL_ID_REVISION_INVALID,
         )
-        with self.assertRaisesRegex(requests.exceptions.HTTPError, "404 Client Error"):
+        with self.assertRaisesRegex(
+            RevisionNotFoundError, "404 Client Error: Revision Not Found"
+        ):
+            _ = cached_download(url, legacy_cache_layout=True)
+
+    def test_repo_not_found(self):
+        # Invalid model file.
+        url = hf_hub_url("bert-base", filename="pytorch_model.bin")
+        with self.assertRaisesRegex(
+            RepositoryNotFoundError, "401 Client Error: Repository Not Found"
+        ):
             _ = cached_download(url, legacy_cache_layout=True)
```

Comment:
Note the behavior changes when authenticated - in that case the error message will be:
`404 Client Error: Repository Not Found`